ci/scripts/python_wheel_xlinux_build.sh

```diff
@@ -124,6 +125,7 @@ cmake \
     -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE} \
     -DCMAKE_INSTALL_LIBDIR=lib \
     -DCMAKE_INSTALL_PREFIX=/tmp/arrow-dist \
+    -DCMAKE_INTERPROCEDURAL_OPTIMIZATION=${CMAKE_INTERPROCEDURAL_OPTIMIZATION} \
```

Comment:
This is for C++, but can we also try passing it for Python using `PYARROW_CMAKE_OPTIONS`?

Reply:
that makes sense, let me have all Arrow C++ builds with it correctly and I will try to add it to pyarrow afterwards